### PR TITLE
EXSWHTEC-291 - Implement tests for floating-point and integer math intrinsics

### PR DESCRIPTION
**Pull request by mirza-halilcevic**
- Accuracy tests for single-precision and double-precision floating-point intrinsics
- Sanity tests for integer intrinsics
- Negative tests for single-precision floating-point, double-precision floating-point, and integer intrinsics.

Depends-on: #260

### CODE REVIEW

File: catch/include/cmd_options.hh [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  int iterations = 1000;
+  int warmups = 100;
+  int cg_extended_run = 5;
+  bool no_display = false;
+  bool progress = false;
+  uint64_t accuracy_iterations = std::numeric_limits<uint32_t>::max() + 1ull;
+  int accuracy_max_memory = 80;
+};
+
+extern CmdOptions cmd_options;
```
Comment (yxsamliu) [inline catch/include/cmd_options.hh]: missing new line

File: catch/include/hip_test_common.hh [hunk 2]
```diff
@@ -112,6 +112,19 @@ THE SOFTWARE.
...
   }
 
+// Check that an expression, errorExpr, evaluates to the expected error_t, expectedError.
+#define HIPRTC_CHECK_ERROR(errorExpr, expectedError)                                               \
+  {                                                                                                \
+    auto localError = errorExpr;                                                                   \
+    INFO("Matching Errors: "                                                                       \
+         << "\n    Expected Error: " << hiprtcGetErrorString(expectedError)                        \
+         << "\n    Expected Code: " << expectedError << '\n'                                       \
+         << "                  Actual Error:   " << hiprtcGetErrorString(localError)               \
```
Comment (yxsamliu) [inline catch/include/hip_test_common.hh:139]: indentation mismatch

Comment (mirza-halilcevic) [reply catch/include/hip_test_common.hh:139]: The clang-format configuration provided in the project formats the code like this. Do you want me to change the indentation in some particular way?

**Comment by rakesroy** [comment]
PR has been merged into develop branch via commit 17291543418b487f6f42e6aa7b4a46adfa48f3b3.

**Pull request closed without merging**